snooty/main.py

```diff
@@ -84,7 +84,12 @@ def on_update(self, prefix: List[str], page_id: FileId, page: Page) -> None:
             asset.get_checksum() for asset in page.static_assets if asset.can_upload()
         )
 
-        fully_qualified_pageid = "/".join(prefix + [page_id.with_suffix("").as_posix()])
+        if "extracts" in page_id.as_posix():
```

Comment:
Mmmmm... this will be quite brittle I think and just masks the underlying problem: that `with_suffix("")` *cannot* deal properly with files that may or may not have an extension.

I'm not sure whether or not we even need that; will a page_id ever have an extension? Try just removing that method call, and see if any tests break. Also try adding `assert page_id.suffix not in {".txt", ".rst", ".yaml"}` and see if it barfs on building some of the cloud repos.

If just removing `with_suffix("")` turns out to not work, another approach would be to just strip `\.(txt)|(rst)|(yaml)$` from the id using `page_id.with_name()` and `page_id.name`